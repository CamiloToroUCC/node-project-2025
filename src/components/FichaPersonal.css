.ficha-personal {
    background-color: #ffffff;
    border: 1px solid #ddd;
    padding: 20px;
    max-width: 600px;
    margin: 30px auto;
    border-radius: 8px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    font-family: 'Arial', sans-serif;
  }
  
  .ficha-personal h1 {
    text-align: center;
    color: #333;
    margin-bottom: 20px;
  }
  
  .ficha-personal img {
    display: block;
    margin: 0 auto 20px;
    border-radius: 50%;
    width: 150px;
    height: 150px;
    object-fit: cover;
  }
  
  .ficha-personal p {
    font-size: 16px;
    color: #555;
    line-height: 1.6;
    text-align: center;
    margin: 10px 0;
  }
  