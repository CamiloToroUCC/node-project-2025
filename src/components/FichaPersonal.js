import React from "react";
import './FichaPersonal.css';

export default function FichaPersonal({ user }) {
  if (!user) return null; // No renderiza nada si no hay usuario seleccionado

  return (
    <div className="ficha-personal" data-testid="ficha-personal">
      <h1>Ficha de Usuario</h1>
      <img src={user.picture.large} alt={`${user.name.first} ${user.name.last}`} />
      <p>Nombre: {user.name.first} {user.name.last}</p>
      <p>Email: {user.email}</p>
      <p>Pa√≠s: {user.location.country}</p>
    </div>
  );
}
