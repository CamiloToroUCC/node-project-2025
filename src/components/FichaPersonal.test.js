import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import FichaPersonal from "./FichaPersonal";

describe("Componente FichaPersonal", () => {
  describe("Cuando el componente se renderiza inicialmente sin datos", () => {
    it("no debe renderizar nada si no se pasa un usuario", () => {
      const { container } = render(<FichaPersonal />);
      expect(container.firstChild).toBeNull();
    });
  });

  describe("Cuando se renderiza con datos predeterminados", () => {
    const defaultUser = {
      picture: { large: "default-image.jpg" },
      name: { first: "John", last: "Doe" },
      email: "john.doe@example.com",
      location: { country: "Estados Unidos" },
    };

    beforeEach(() => {
      render(<FichaPersonal user={defaultUser} />);
    });

    it('debe mostrar el título "Ficha de Usuario" con el rol "heading"', () => {
      expect(screen.getByRole("heading", { name: /Ficha de Usuario/i })).toBeInTheDocument();
    });
  
    it('debe renderizar un párrafo que contenga el texto "Nombre: John Doe"', () => {
      expect(screen.getByText(/Nombre: John Doe/i)).toBeInTheDocument();
    });
  
    it('debe renderizar un párrafo que contenga el texto "Email: john.doe@example.com"', () => {
      expect(screen.getByText(/Email: john\.doe@example\.com/i)).toBeInTheDocument();
    });
  
    it('debe renderizar un párrafo que contenga el texto "País: Estados Unidos"', () => {
      expect(screen.getByText(/País: Estados Unidos/i)).toBeInTheDocument();
    });
  
    it('debe tener el atributo data-testid "ficha-personal" en el contenedor principal', () => {
      expect(screen.getByTestId("ficha-personal")).toBeInTheDocument();
    });
  });

  describe("Con diferentes datos de usuario (simulado)", () => {
    const userData = {
      picture: { large: "jane-image.jpg" },
      name: { first: "Jane", last: "Smith" },
      email: "jane.smith@example.com",
      location: { country: "Canadá" }
    };

    beforeEach(() => {
      render(<FichaPersonal user={userData} />);
    });

    it("debe mostrar el nombre correcto para otro usuario", () => {
      expect(screen.getByText(/Nombre: Jane Smith/i)).toBeInTheDocument();
      expect(screen.queryByText(/Nombre: John Doe/i)).toBeNull();
    });

    it("debe mostrar el email correcto para otro usuario", () => {
      expect(screen.getByText(/Email: jane\.smith@example\.com/i)).toBeInTheDocument();
      expect(screen.queryByText(/Email: john\.doe@example\.com/i)).toBeNull();
    });

    it("debe mostrar el país correcto para otro usuario", () => {
      expect(screen.getByText(/País: Canadá/i)).toBeInTheDocument();
      expect(screen.queryByText(/País: Estados Unidos/i)).toBeNull();
    });
  });

  describe("Prueba extendida de interacción (opcional)", () => {
    const TestFichaPersonalInteractiva = ({ onClose, user }) => (
      <div className="ficha-personal" data-testid="ficha-personal">
        <h1>Ficha de Usuario</h1>
        <button onClick={onClose} data-testid="close-button">Cerrar</button>
        <p>Nombre: {user.name.first} {user.name.last}</p>
      </div>
    );

    it("debe ejecutar una acción cuando se clickea el botón 'Cerrar'", () => {
      const mockOnClose = jest.fn();
      const userData = {
        picture: { large: "jane-image.jpg" },
        name: { first: "Jane", last: "Smith" },
        email: "jane.smith@example.com",
        location: { country: "Canadá" }
      };

      render(<TestFichaPersonalInteractiva onClose={mockOnClose} user={userData} />);
      const closeButton = screen.getByTestId("close-button");
      fireEvent.click(closeButton);
      expect(mockOnClose).toHaveBeenCalledTimes(1);
    });
  });
});
