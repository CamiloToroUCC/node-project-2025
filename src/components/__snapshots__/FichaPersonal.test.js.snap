// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente FichaPersonal Pruebas de regresión debe coincidir con el snapshot 1`] = `
<DocumentFragment>
  <div
    class="ficha-personal"
    data-testid="ficha-personal"
  >
    <h1>
      Ficha de Usuario
    </h1>
    <p>
      Nombre: John Doe
    </p>
    <p>
      Email: john.doe@example.com
    </p>
    <p>
      País: Estados Unidos
    </p>
  </div>
</DocumentFragment>
`;

exports[`Pruebas para FichaPersonal FichaPersonal coincide con el snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Random People
    </h1>
    <div
      class="SearchForm"
    >
      <div
        class="SearchForm-field"
      >
        <select>
          <option
            value=""
          >
            Both
          </option>
          <option
            value="male"
          >
            Male
          </option>
          <option
            value="female"
          >
            Female
          </option>
        </select>
      </div>
      <div
        class="SearchForm-field"
      >
        <input
          placeholder="Select country code: us, it,..."
          type="text"
          value="US"
        />
      </div>
    </div>
    <div
      class="App-controls"
    >
      <button
        class="btn"
      >
        Buscar con Axios
      </button>
      <button
        class="btn"
      >
        Buscar con Fetch
      </button>
      <button
        class="btn"
      >
        Comparar Axios vs Fetch
      </button>
      <button
        class="btn"
      >
        Ficha Personal
      </button>
    </div>
    <div
      class="ficha-personal"
      data-testid="ficha-personal"
    >
      <h1>
        Ficha de Usuario
      </h1>
      <p>
        Nombre: John Doe
      </p>
      <p>
        Email: john.doe@example.com
      </p>
      <p>
        País: Estados Unidos
      </p>
    </div>
  </div>
</DocumentFragment>
`;
